src/QuadForm/Quad/ZLattices.jl

```diff
@@ -391,30 +396,53 @@ function assert_has_automorphisms(L::ZZLat; redo::Bool = false,
     return nothing
   end
 
+  _alpha, sv = known_short_vectors
   V = ambient_space(L)
   GL = gram_matrix(L)
   d = denominator(GL)
-  res = ZZMatrix[change_base_ring(ZZ, d * GL)]
+  if !isone(d)
+    res = ZZMatrix[change_base_ring(ZZ, d * GL)]
+    if !iszero(_alpha)
+      alpha = abs(numerator(d*_alpha))
+      sv = eltype(sv)[(v[1], d*v[2]) for v in sv]
+    else
+      alpha = _alpha
+    end
+  else
+    res = ZZMatrix[change_base_ring(ZZ, GL)]
+    if !iszero(_alpha)
+      alpha = abs(numerator(_alpha))
+    else
+      alpha = _alpha
+    end
+  end
+
   # So the first one is either positive definite or negative definite
   # Make it positive definite. This does not change the automorphisms.
   if res[1][1, 1] < 0
     res[1] = -res[1]
   end
-  # Make the Gram matrix small
-  Glll, T = lll_gram_with_transform(res[1])
-  res[1] = Glll
+  if !get_attribute(L, :is_lll_reduced, false)
+    # Make the Gram matrix small
+    Glll, T = lll_gram_with_transform(res[1])
+    res[1] = Glll
+    sv = eltype(sv)[(solve(T, v[1]; side=left), v[2]) for v in sv]
```

Comment:
This need a solve_init because you solve with the same left hand side many times. 
But in any case: is this faster than recomputing the short vectors?

Reply:
Please use the solve init anyways :-). Otherwise, it will put T in hermite normal form a thousand times.